packages/cli-lib/lib/handleFieldsJs.js

```diff
@@ -0,0 +1,116 @@
+const fsExtra = require('fs-extra');
+const path = require('path');
+const escapeRegExp = require('./escapeRegExp');
+const { logger } = require('../logger');
+const { getCwd } = require('@hubspot/cli-lib/path');
+const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const { getExt, splitLocalPath } = require('../path');
+const i18nKey = 'cli.commands.upload';
+
+function handleFieldErrors(e, filePath) {
+  if (e instanceof SyntaxError) {
+    const ext = getExt(filePath);
+    if (ext == 'js') {
+      logger.error(i18n(`${i18nKey}.errors.jsSyntaxError`, { js: filePath }));
+    }
+  }
+  if (e.code === 'ENOENT') {
+    logger.error(
+      i18n(`${i18nKey}.errors.invalidPath`, {
+        path: filePath,
+      })
+    );
+  }
+  if (e.code === 'MODULE_NOT_FOUND') {
+    logger.error(
+      i18n(`${i18nKey}.errors.jsSyntaxError`, {
+        path: filePath,
+      })
+    );
+  }
+}
+
+/*
+ * Determines if file is a processable fields.js file (i.e., if it is in the root or in a module folder)
+ */
+function isProcessableFieldsJs(src, filePath) {
+  const regex = new RegExp(`^${escapeRegExp(src)}`);
+  const parts = splitLocalPath(filePath);
+  const relativePath = filePath.replace(regex, '');
+  const baseName = path.basename(filePath);
+  const moduleFolder = parts.find(part => part.endsWith('.module'));
+  return (
+    baseName == 'fields.js' && (moduleFolder || relativePath == '/fields.js')
+  );
+}
+
+/**
+ * Converts a fields.js file into a fields.json file, writes, and returns of fields.json
+ * @param {string} file - The path of the fields.js javascript file.
+ * @param {string[]} options - Optional arguments to pass to the exported function in fields.js
+ * @param {string} writeDir - The path to write the file to.
+ * @returns {string} The path of the written fields.json file.
+ */
+async function convertFieldsJs(filePath, options, writeDir) {
+  const dirName = path.dirname(filePath);
+  logger.info(
+    i18n(`${i18nKey}.converting`, {
+      src: dirName + '/fields.js',
+      dest: dirName + '/fields.json',
+    })
+  );
+
+  try {
+    /* Since we require() fields.js files, node will cache them.
+     * Thus during an hs watch, if the user edits a fields.js file, node will return the cached version on require() instead of re-requiring the new one.
+     * So we clean the cache before requiring so that node requires the fresh fields.js. See https://nodejs.org/api/modules.html#caching.
+     */
+    delete require.cache[filePath];
+
+    // Save CWD and then switch CWD to the project. This is so that any calls to the file system that are written relatively will resolve properly.
+    const cwd = getCwd();
+    process.chdir(dirName);
```

Comment:
Is the risk here that `filePath` might be relative? I'm wondering if we could use `path.resolve()` where needed to make sure we're working with absolute paths and avoid changing the cwd.

Reply:
The comment is probably a bit unclear, sorry. 

By "any calls to the filesystem" I mean calls that are in the `fields.js`. So in the case when we don't change the cwd, if a dev has `fs.readFileSync('../example.json')` in their `fields.js`, then we execute the readFile and look for the `example.json` file relative to whatever the cwd of the shell is. So for a `fields.js` in a module folder, this wouldn't work unless they cd directly into that folder. 

The alternative solution to this problem was providing file reading proxy functions or doing some call-stack trickery to resolve the path of the module that they are requiring, but both of these seem scarier than just changing the process directory while we load the module.

I have a feeling there is a better way to do this built into node but I was never able to find a good solution besides chdir.

Reply:
Ah, very good point- I hadn't considered the actual `fields.js` resolution. Seems like a reasonable way to handle this